From e178527793d744e2d1d5728317e7d550c4183eaa Mon Sep 17 00:00:00 2001
From: hudeng <hudeng@deepin.org>
Date: Wed, 20 Dec 2023 07:56:59 +0000
Subject: [PATCH] chore(obs): backend build-dsc script add eatmydata support
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

同时riscv64和loong64跳过某些构建选项
---
 build-recipe-dsc | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/build-recipe-dsc b/build-recipe-dsc
index b9b26af..a7fff10 100644
--- a/build-recipe-dsc
+++ b/build-recipe-dsc
@@ -151,7 +151,15 @@ dsc_build() {
     # and alternate build commands can be used
     DSC_BUILD_CMD="$(queryconfig --dist "$BUILD_DIST" --archpath "$BUILD_ARCH" --configdir "$CONFIG_DIR" substitute dsc:build_cmd)"
     test -z "$DSC_BUILD_CMD" && DSC_BUILD_CMD="dpkg-buildpackage -us -uc"
+    if [ -n "$(command -v eatmydata)" ];then echo "Using eatmydata" && DSC_BUILD_CMD="eatmydata $DSC_BUILD_CMD";fi
+    if [ "$BUILD_ARCH" = "riscv64" ];then
+        DSC_BUILD_CMD="$DSC_BUILD_CMD --no-check-builddeps --build-profiles=nocheck"
+    fi
+
+    if [ "$BUILD_ARCH" = "loong64" ];then
+        DSC_BUILD_CMD="$DSC_BUILD_CMD --no-check-builddeps --build-profiles=nocheck"
+    fi

     if grep -Eq '^Format: 3\.0 \(quilt\)$' $BUILD_ROOT$TOPDIR/SOURCES/$RECIPEFILE ; then
         printf "Recipe file '$RECIPEFILE' is being checked for 'DEBTRANSFORM-FILES'... "
         if grep -Eq '^DEBTRANSFORM-FILES:' $BUILD_ROOT$TOPDIR/SOURCES/$RECIPEFILE ; then
-- 
2.35.3
